### ISSUE TITLE
some images are not rendered

### ISSUE THREAD

**Issue by giuseppebianchi** [robertjdominguez/docusaurus-to-pdf#1]
some images are not present in the final pdf

**Comment by robertjdominguez** [comment]
@giuseppebianchi — images are rendered using [this](https://github.com/robertjdominguez/docusaurus-to-pdf/blob/main/src/generator/utils.ts#L40-L53) concatenation:

```ts
export function resolveImageUrls(html: string, baseUrl: string): string {
  const $ = cheerio.load(html);

  $("img").each((_, img) => {
    const src = $(img).attr("src");
    if (src && !src.startsWith("http")) {
      // Convert relative URLs to absolute URLs
      const absoluteUrl = new URL(src, baseUrl).href;
      $(img).attr("src", absoluteUrl);
    }
  });

  return $.html();
}
```

Is it safe to say your unrendered images have a different `baseUrl`? If so, feel free to open a PR to handle parsing this differently.

**Comment by giuseppebianchi** [comment]
I checked the images urls and everything was ok. While debugging with headless mode disabled, I identified the issue: it's related to the 'lazy loading' behavior, so images outside the visibile viewport are not included into the generated pdf.
I've implemented an utility to force image loading, which can be enabled via cli, and now all images of my documentation are properly displayed in the pdf. 😊
I'll open a PR.
